src/vehicle.cpp

```diff
@@ -273,7 +273,7 @@ bool Vehicle::NeedsAutomaticServicing() const
 {
 	if (this->HasDepotOrder()) return false;
 	if (this->current_order.IsType(OT_LOADING)) return false;
-	if (this->current_order.IsType(OT_GOTO_DEPOT) && this->current_order.GetDepotOrderType() != ODTFB_SERVICE) return false;
+	if (this->current_order.IsType(OT_GOTO_DEPOT) && this->current_order.GetDepotOrderType() == ODTF_MANUAL) return false;
```

Comment:
The current comparison can be considered wrong, but I would say the new variant is more wrong.

I'm more looking to the future here; lets assume a new flag gets introduced. Do we want the behaviour of that flag to also include "please reconsider the depot for automatic servicing"? Or should it be that the new flag does not affect the automatic servicing behaviour?
Of course the comparison can be changed at that moment, but the fact that `ODTFB_PART_OF_ORDER` won't ever be set here is hidden far away. That makes determining what the intention of the comparison is obtuse, and makes thinking about what the conditions should be harder. So I rather let the comparison show the actual intentions.

So, should this function read "do reconsider servicing if and only if `ODTFB_SERVICE` is set"?
```suggestion
	if (this->current_order.IsType(OT_GOTO_DEPOT) && (this->current_order.GetDepotOrderType() & ODTFB_SERVICE) == 0) return false;
```